schemas/chosen.yaml

```diff
@@ -86,13 +86,15 @@ properties:
   linux,usable-memory-range:
     allOf:
       - $ref: types.yaml#definitions/uint64-array
-      - maxItems: 2
+      - maxItems: 4
```

Comment:
I don't think we need to change the limit to 2 ranges, but allow for N ranges. The type should be changed to uint64-matrix with the constraints like this:

items:
  maxItems: 2

Reply:
Ok, allow for N ranges and then limit in kernel.
I take a look at the memory nodes, if we could also do like the way you mentioned here. 
`properties:
  $nodename:
    pattern: "^memory(@[0-9a-f]+)?$"
  device_type:
    const: memory
  reg:
    minItems: 1
    maxItems: 1024`

Reply:
Ok, allow for N ranges and then limit in kernel.
I take a look at the memory nodes, in the memory nodes, if we could also do like the way you mentioned above.
```
properties:
  $nodename:
    pattern: "^memory(@[0-9a-f]+)?$"
  device_type:
    const: memory
  reg:
    minItems: 1
    maxItems: 1024
```